sdk/healthcareapis/Azure.ResourceManager.HealthcareApis/tests/HealthcareApisManagementTestBase.cs

```diff
@@ -30,12 +35,69 @@ public void CreateCommonClient()
+        protected async Task<ResourceGroupResource> CreateResourceGroup()
         {
-            string rgName = Recording.GenerateAssetName(rgNamePrefix);
-            ResourceGroupData input = new ResourceGroupData(location);
+            SubscriptionResource subscription = await Client.GetDefaultSubscriptionAsync();
+            string rgName = Recording.GenerateAssetName(ResourceGroupNamePrefix);
+            ResourceGroupData input = new ResourceGroupData(DefaultLocation);
             var lro = await subscription.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, rgName, input);
             return lro.Value;
         }
+
+        protected async Task<HealthcareApisWorkspaceResource> CreateHealthcareApisWorkspace(ResourceGroupResource resourceGroup, string workspaceName)
+        {
+            var data = new HealthcareApisWorkspaceData(DefaultLocation);
+            var lro = await resourceGroup.GetHealthcareApisWorkspaces().CreateOrUpdateAsync(WaitUntil.Completed, workspaceName, data);
+            return lro.Value;
+        }
+
+        protected async Task<FhirServiceResource> CreateFhirService(HealthcareApisWorkspaceResource workspace, string fhirServiceName)
+        {
+            FhirServiceData data = new FhirServiceData(DefaultLocation)
+            {
+                Kind = "fhir-R4",
+                AuthenticationConfiguration = new FhirServiceAuthenticationConfiguration()
+                {
+                    Authority = $"https://login.microsoftonline.com/{Environment.GetEnvironmentVariable("TENANT_ID")}",
+                    Audience = $"https://{workspace.Data.Name}-{fhirServiceName}.fhir.azurehealthcareapis.com"
+                },
+                ImportConfiguration = new FhirServiceImportConfiguration()
+                {
+                    IsEnabled = false,
+                    IsInitialImportMode = false,
+                },
+                ResourceVersionPolicyConfiguration = new FhirServiceResourceVersionPolicyConfiguration()
+                {
+                    Default = "no-version",
+                }
+            };
+            var lro = await workspace.GetFhirServices().CreateOrUpdateAsync(WaitUntil.Completed, fhirServiceName, data);
+            return lro.Value;
+        }
+
+        protected async Task<HealthcareApisIotConnectorResource> CreateHealthcareApisIotConnector(ResourceGroupResource resourceGroup, HealthcareApisWorkspaceResource workspace, string iotConnectorName)
+        {
+            // Create an Eventhub
+            var eventhubNamespace = await resourceGroup.GetEventHubsNamespaces().CreateOrUpdateAsync(WaitUntil.Completed, Recording.GenerateAssetName("azmedtechspaces"), new EventHubsNamespaceData(resourceGroup.Data.Location));
+            string eventHubName = Recording.GenerateAssetName("eventhub");
+            var eventhub = await eventhubNamespace.Value.GetEventHubs().CreateOrUpdateAsync(WaitUntil.Completed, eventHubName, new EventHubData());
```

Comment:
We need to exclude the API calls from other RPs from the recording to avoid the case in the future, api-version of event hub upgraded and we have to record the test case in this RP again.

Reply:
Added a new rule to UriRegexSanitizers in the constructor of TestBase to ignore event hubs api-version